### PR TITLE
Rewrite errors observable

### LINKED ISSUE #63: Actions defaults to enabled=false
I have updated to 2.1.0 recently and im experiencing that some actions not enabled by default.
This means i immediately get an error `notEnabled`, but the work factory seems to execute. Theres never a call back to my subscription on elements. 

I looked around in the recent changes and this seems peculiar:
```
    public init(
        enabledIf: Observable<Bool> = Observable.just(true),
        workFactory: @escaping WorkFactory) {
        
        self._enabledIf = enabledIf
        self.workFactory = workFactory

        let enabledSubject = BehaviorSubject<Bool>(value: false)
        enabled = enabledSubject.asObservable()
```
I'm not 100% sure whats going on, but it seems to me `_enabledIf` defaults true and `enabled` defaults to `false`?

### PR DESCRIPTION
**Pull request by fpillet**
This PR rewrites both the creation of Action's `error` observable inside the Action itself and restructures error handling in the `execute` function.

The motivation is to try and fix #63. The only plausible reason that could explain this issue is that the `notEnabledError` observable previously created was derived from `inputs`. Since `executionObservable` (the root observable for all other work inside action) was derived from `inputs` too, one couldn't guarantee the order of processing on `input` update, depending on who subscribed to what and in what order.

To fix the issue, I decided to have a single point of contact with `inputs`, and derive everything from it. This led to some not-perfectly-Rx-y code where I push errors to an internal subject using side effects, but it really is less important than getting Action to perform right in all circumstances.

I still have to find a way to reproduce issue #63, waiting on @tomburns to confirm whether this change fixes his issues.

### CODE REVIEW

**Comment by fpillet** [comment]
Ok so @tomburns confirmed that the fix indeed resolves #63. Looks like we're good to go with this one.

**Comment by tomburns** [comment]
Regarding a repro and possible test case for this: 

In our case, we were doing the following, all in the same method:

1. instantiating the Action
2. subscribing to `elements`
3. subscribing to `errors`
4. passing the Action off to its consumer.

File: Sources/Action/Action.swift [hunk 1]
```diff
@@ -122,23 +107,20 @@ public final class Action<Input, Element> {
...
-            .of(execution, notEnabledError)
-            .merge()
-            .subscribe(subject)
-            .addDisposableTo(disposeBag)
-
-        return subject.asObservable()
+		let subject = ReplaySubject<Element>.createUnbounded()
+		
+		let work = executionObservables
+			.map { $0.catchError { throw ActionError.underlyingError($0) } }
```
Comment (ishkawa) [inline Sources/Action/Action.swift]: Is `throw ActionError.underlyingError($0)` necessary? 
The error seems to appear in errors subject too.

If that is true, `catchError { _ in Observable.empty() }` is more suitable.

File: Sources/Action/Action.swift [hunk 2]
```diff
@@ -61,13 +61,19 @@ public final class Action<Input, Element> {
...
+        errors = errorsSubject.asObservable()
         
         executionObservables = inputs
             .withLatestFrom(enabled) { $0 }
             .flatMap { input, enabled -> Observable<Observable<Element>> in
                 if enabled {
-                    return Observable.of(workFactory(input).shareReplay(1))
+                    return Observable.of(workFactory(input)
+                                             .do(onError: { errorsSubject.onNext(.underlyingError($0)) })
+                                             .shareReplay(1))
```
Comment (ishkawa) [inline Sources/Action/Action.swift]: nit: Introducing variable for the execution observable makes this part more readable.

```swift
let execution = workFactory(input)
    .do(onError: { errorsSubject.onNext(.underlyingError($0)) })
    .shareReplay(1)

return Observable.of(execution)
```

Comment (fpillet) [inline Sources/Action/Action.swift]: I don't see how this makes the code more readable. Actually I think it makes it _less_ readable

File: Sources/Action/Action.swift (hunk 1 above)
Comment (fpillet) [inline Sources/Action/Action.swift]: If you look at the previous code, it was doing exactly this -- propagate errors to the execution observable returned by the `execute()` function, both for errors produced by the execution, and for the `notEnabled` error. I kept the same behaviour.

File: Sources/Action/Action.swift (hunk 2 above)
Comment (ishkawa) [inline Sources/Action/Action.swift]: OK. Let's leave it as it is.

File: Sources/Action/Action.swift (hunk 1 above)
Comment (ishkawa) [inline Sources/Action/Action.swift]: I got it 👍

**Comment by ashfurrow** [comment]
I'll try to get this released tomorrow, thanks again 👍

**Comment by ashfurrow** [comment]
Okay, all released for CocoaPods and Carthage!

**Comment by fpillet** [comment]
Thanks, Ash! 👏

**Pull request merged**